### PR TITLE
auto_init: some more cleanup and unification

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
I only looked at #13364 after the fact and found some more ways to improve it:
1. Unify the last remaining `DEBUG()` (outputting `"auto_init 6LoWPAN border router DHCPv6 client"`) with the rest and use `LOG_DEBUG()`. Consequently, remove the `debug.h` include
2. Unify how functions are included to use `extern` (to only require one `#ifdef MODULE_…` and consequently remove all unneeded `#include`s
3. Move grouped initializations to their own functions, see https://github.com/RIOT-OS/RIOT/pull/13542#discussion_r387118914 and https://github.com/RIOT-OS/RIOT/pull/13542#discussion_r387119308
3. Use `IS_USED(MODULE_…)` instead of `#ifdef MODULE_` to compile-time check 1., 2., and 3.

### Testing procedure
All applications should still compile, auto initialization should still work (running some tests on different boards and `examples/default` and `examples/gnrc_network` should be enough to show that)

### Issues/PRs references
Follow-up on #13364.

### CODE REVIEW

**Comment by miri64** [comment]
I don't insist of the `IS_USED()` commit, but I think it provides a benefit.

File: sys/auto_init/auto_init.c [hunk 1]
```diff
@@ -16,653 +16,580 @@
...
+
+        if (IS_USED(MODULE_SX127X) && !IS_USED(MODULE_SEMTECH_LORAMAC)) {
+            extern void auto_init_sx127x(void);
+            auto_init_sx127x();
+        }
+
+        if (IS_USED(MODULE_NRF802154)) {
+            extern void auto_init_nrf802154(void);
+            auto_init_nrf802154();
+        }
```
Comment (benpicco) [inline sys/auto_init/auto_init.c]: Can you move those to a separate function while you're at it, so manual init is possible in a generic way?

So you'd have

```C
if (IS_USED(MODULE_AUTO_INIT_GNRC_NETIF)) {
    gnrc_netif_init();
}
```

Where `gnrc_netif_init()` will take care of calling the individual netif init functions.

File: sys/auto_init/auto_init.c [hunk 2]
```diff
@@ -16,653 +16,580 @@
...
+            auto_init_tsl4531x();
+        }
+        if (IS_USED(MODULE_VCNL40X0)) {
+            extern void auto_init_vcnl40x0(void);
+            auto_init_vcnl40x0();
+        }
+        if (IS_USED(MODULE_VEML6070)) {
+            extern void auto_init_veml6070(void);
+            auto_init_veml6070();
+        }
```
Comment (benpicco) [inline sys/auto_init/auto_init.c]: Same can be done here for SAUL.

File: sys/auto_init/auto_init.c (hunk 1 above)
Comment (miri64) [reply sys/auto_init/auto_init.c]: Needs some restructuring of my commit order, but will do.

Comment (miri64) [reply sys/auto_init/auto_init.c]: Addressed.

File: sys/auto_init/auto_init.c (hunk 2 above)
Comment (miri64) [reply sys/auto_init/auto_init.c]: Addressed

**Comment by miri64** [comment]
Addressed. I needed to force-push however, since I wanted to also have a variant with `#ifdef` of the grouping functions in the history ;-).

**Comment by miri64** [comment]
Huh??!? Somehow the IS_USED commit got mixed up with the function PR.
Lemme rebase this correctly.

**Comment by miri64** [comment]
Fixed

**Comment by miri64** [comment]
> Cleanup looks good, just moves code around.

You should look again. There is also a functional change... (moving the call conditions from the preprocessor to the compiler/optimizer ;-).

**Comment by benpicco** [comment]
> There is also a functional change... (moving the call conditions from the preprocessor to the compiler/optimizer ;-)

That should result in the same code being generated though…

**Comment by miri64** [comment]
> > There is also a functional change... (moving the call conditions from the preprocessor to the compiler/optimizer ;-)
> 
> That should result in the same code being generated though…

Yepp :-) still not just moving code around ;-).

**Comment by benpicco** [comment]
[`run_test/tests/nordic_softdevice/nrf52dk:gnu`](https://ci.riot-os.org/RIOT-OS/RIOT/13542/6765ff8a77a5eb40ebc67dd81ff9aecdc49b08e1/output/run_test/tests/nordic_softdevice/nrf52dk:gnu.txt) and [`run_test/examples/suit_update/nrf52dk:gnu`](https://ci.riot-os.org/RIOT-OS/RIOT/13542/6765ff8a77a5eb40ebc67dd81ff9aecdc49b08e1/output/run_test/examples/suit_update/nrf52dk:gnu.txt) are currently *always* failing.

**Comment by gschorcht** [comment]
> I don't insist of the `IS_USED()` commit, but I think it provides a benefit.

Oops, I was really surprised about this change when I tried to rebase a driver module and resolve the conflict. What exactly is the benefit of `IS_USED` over `#ifdef MODULE_` in this context with the exception of the compile time check for the external function declaration?

While the `#ifdef MODULE_` allows to compile out the parts that aren't used at all, `IS_USED` at least generates the code for the `if` statements for all modules. I'm not sure whether all compilers are able to optimize out the `if` block if the condition is always false. Did you checked the code size before and after this change?

Another small problem is that I have dozens of PRs open for which conflict resolution is no longer trivial.

**Comment by gschorcht** [comment]
@mir64 Forget my comment. I crawled through the discussion of PR #12626 and the new coding convention in PR #12781 I did not yet realized :wink: OK, I will use `IS_USED` from now wherever it is possible :sunglasses: But the little problem with resolving the conflicts of open PRs remains :worried:

**Comment by miri64** [comment]
Sorry :-(, I guess that is a problem with major reworks in general :sweat_smile:

**Pull request merged**